ballerina/client.bal

```diff
@@ -256,6 +256,78 @@ public isolated client class Client {
         http:Response httpResponse = check self.amazonS3->delete(requestURI, request);
         return handleHttpResponse(httpResponse);
     }
+    
+    # Generates a presigned URL for the object.
+    #
+    # + bucketName - The name of the bucket  
+    # + objectName - The name of the object  
+    # + action -   The HTTP method or the HTTP method with respective headers(objectCreationHeaders or objectRetrievalHeaders)
+    # + expires - The time period for which the presigned URL is valid, in seconds  
+    # + partNo - The part number of the object, when uploading multipart objects  
+    # + uploadId - The upload ID of the multipart upload
+    # + return - If successful, a presigned URL, else an error
+    @display {label: "Create Presigned URL"}
+    remote isolated function createPresignedUrl(
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "Object Name"} string objectName,
+            @display {label: "HTTP Method and Optional Headers"} GET|PUT|GetHeaders|PutHeaders action,
```

Comment:
On second thoughts, shall we just use

```ballerina
GET|PUT|ObjectCreationHeaders|ObjectRetrievalHeaders
```

?

The reason to not suggest this previously was because the type won't get inferred between ObjectCreationHeaders and ObjectRetrievalHeaders with a mapping constructor. But they can still use a cast or a separate variable.

```ballerina
<ObjectRetrievalHeaders> {...}
``` 

which is OK IMO. Since even with the current approach we have to do `{objectRetrievalHeaders: {...}}`.

Reply:
@MaryamZi this approach should work, but not sure if the union of `GET|PUT|ObjectCreationHeaders|ObjectRetrievalHeaders` might be confusing to the users, as two of the union types represent only the method and, while the other two types represent headers. 

In order to address this case, can't we define the records using type inclusions to represent the latter case as below?

```ballerina
public type PUTMethodWithHeaders record {
    *ObjectCreationHeaders;
    PUT method = PUT; 
};

public type GETMethodWithHeaders record {
   *ObjectRetrievalHeaders;
   GET method = GET; 
};
```

then we won't have to set headers as `{objectRetrievalHeaders: {...}}` either.

Reply:
Thinking about this more, we don't really need the method, right? It's more the action. How about

```
CREATE|RETRIEVE|ObjectCreationHeaders|ObjectRetrievalHeaders
```

?

Reply:
@MaryamZi What if leave the method parameter as is and have another optional parameter as ObjectCreationalHeaders | ObjectRetrievalHeaders, basically
```Ballerina
GET|PUT httpMethod,
ObjectCreationHeaders|ObjectRetrievalHeaders headers = ()
```
and check within the function if the right  type of header has been passed?

I feel like this would be more intuitive for users

Reply:
The problem with that is that we will be giving a runtime error for something we can detect at compile-time. We're basically opening up room for error.

Reply:
I'm also more towards to maryam's last suggestion as 

> CREATE|RETRIEVE|ObjectCreationHeaders|ObjectRetrievalHeaders

Yes this makes more sense than combining HTTP method names and headers into a single union type. @nisuraaa Shall we proceed with this?

Reply:
Sure

Reply:
> ```
> CREATE|RETRIEVE|ObjectCreationHeaders|ObjectRetrievalHeaders
> ```

So in this case CREATE and RETRIEVE will be the strings "create" and "retrieve" ?

Reply:
Yeah, that should do I think.